sys/log/full/include/log/log.h

```diff
@@ -311,11 +301,23 @@ STATS_SECT_END
 
 /* Trailer support callbacks */
 struct log_trailer_handler {
+    /* Trailer length callback used to get the length of the trailer from provided
+     * log entry
+     */
     log_trailer_len_func_t *log_trailer_len;
-    log_trailer_data_len_func_t *log_trailer_data_len;
+    /* Trailer append callback used to append trailer to the log entry */
     log_trailer_append_func_t *log_trailer_append;
+    /* Trailer process callback used to process trailer from the log entry
+     * which gets called at various places where the log entry gets read
+     */
     log_process_trailer_func_t *log_process_trailer;
+    /* Trailer mbuf append callback used to append trailer to the log entry
+     * using the mbufs
+     */
     log_trailer_mbuf_append_func_t *log_trailer_mbuf_append;
+    /* Trailer reset data callback used to reset the trailer data
+     * per log which is defined by the application registering these callbacks
+     */
```

Comment:
As I mentioned in some other comment, there's no "user data" in the callback registration API.